numpy/typing/tests/data/reveal/ma.pyi

```diff
@@ -358,6 +360,12 @@ assert_type(MAR_2d_f4.baseclass, type[NDArray[Any]])
 assert_type(MAR_b.swapaxes(0, 1), MaskedArray[np.bool])
 assert_type(MAR_2d_f4.swapaxes(1, 0), MaskedArray[np.float32])
 
+assert_type(MAR_2d_f4[AR_i8], MaskedArray[np.float32])
+assert_type(MAR_2d_f4[[1, 2, 3]], MaskedArray[np.float32])
+assert_type(MAR_2d_f4[1:], MaskedArray[np.float32])
+assert_type(MAR_2d_V['field_0'], np.ma.MaskedArray[tuple[int, int], np.dtype])
+assert_type(MAR_2d_V[['field_0', 'field_1']], np.ma.MaskedArray[tuple[int, int], np.dtype[np.void]])
```

Comment:
things like `[0, 0]`, `[:]`, `[:, np.newaxis]`, and `[..., -1]` might also be interesting to check